src/test/java/org/opensearch/ad/ml/EntityColdStarterTests.java

```diff
@@ -458,4 +509,258 @@ public void testEmptyDataRange() throws InterruptedException {
+                settings,
+                seed
+            );
+
+            // create labelled data
+            MultiDimDataWithTime dataWithKeys = LabelledAnomalyGenerator
+                .getMultiDimData(
+                    dataSize + detector.getShingleSize() - 1,
+                    50,
+                    100,
+                    5,
+                    seed,
+                    baseDimension,
+                    false,
+                    trainTestSplit,
+                    delta,
+                    anomalyIndependent
+                );
+
+            long[] timestamps = dataWithKeys.timestampsMs;
+            double[][] data = dataWithKeys.data;
+            when(clock.millis()).thenReturn(timestamps[trainTestSplit - 1]);
+
+            doAnswer(invocation -> {
+                ActionListener<Entry<Optional<Long>, Optional<Long>>> listener = invocation.getArgument(2);
+                listener.onResponse(new SimpleImmutableEntry<>(Optional.of(timestamps[0]), Optional.of(timestamps[trainTestSplit - 1])));
+                return null;
+            }).when(searchFeatureDao).getEntityMinMaxDataTime(any(), any(), any());
+
+            doAnswer(invocation -> {
+                List<Entry<Long, Long>> ranges = invocation.getArgument(1);
+                List<Optional<double[]>> coldStartSamples = new ArrayList<>();
+
+                Collections.sort(ranges, new Comparator<Entry<Long, Long>>() {
+                    @Override
+                    public int compare(Entry<Long, Long> p1, Entry<Long, Long> p2) {
+                        return Long.compare(p1.getKey(), p2.getKey());
+                    }
+                });
+                for (int j = 0; j < ranges.size(); j++) {
+                    Entry<Long, Long> range = ranges.get(j);
+                    Long start = range.getKey();
+                    int valueIndex = searchInsert(timestamps, start);
+                    coldStartSamples.add(Optional.of(data[valueIndex]));
+                }
+
+                ActionListener<List<Optional<double[]>>> listener = invocation.getArgument(4);
+                listener.onResponse(coldStartSamples);
+                return null;
+            }).when(searchFeatureDao).getColdStartSamplesForPeriods(any(), any(), any(), anyBoolean(), any());
+
+            String entityNamez = entityName + z;
+            String modelId = detector.getDetectorId() + "_entity_" + entityNamez;
+            EntityModel model = new EntityModel(modelId, new ArrayDeque<>(), null, null);
+            ModelState<EntityModel> modelState = new ModelState<>(
+                model,
+                modelId,
+                detector.getDetectorId(),
+                ModelType.ENTITY.getName(),
+                clock,
+                priority
+            );
+
+            released = new AtomicBoolean();
+
+            inProgressLatch = new CountDownLatch(1);
+            listener = ActionListener.wrap(() -> {
+                released.set(true);
```

Comment:
I don't think we need this listener, atomicBoolean and CountDownLatch. In current code our `trainModel()` utilizes a listener to know if to pull another request from cold start queue which I assume is why we could have used a latchCountDown to release threads being used by the listener in this loop but I don't see it in 1.0 code.

Reply:
you are right. Removed those 3 variables.